tests/components/zha/conftest.py

```diff
@@ -1,72 +1,86 @@
-from homeassistant.components.zha.core.store import async_get_registry
-from homeassistant.helpers.device_registry import async_get_registry as get_dev_reg
+from homeassistant import config_entries
+import homeassistant.components.zha.core.const as zha_const
+import homeassistant.components.zha.core.registries as zha_regs
 
-from .common import FakeDevice, FakeEndpoint, async_setup_entry
-
-from tests.common import MockConfigEntry
+from .common import FakeDevice, FakeEndpoint, get_zha_gateway
 
 FIXTURE_GRP_ID = 0x1001
 FIXTURE_GRP_NAME = "fixture group"
 
 
+@pytest.fixture
+def zigpy_app_controller():
+    """Zigpy ApplicationController fixture."""
+    app = mock.MagicMock(spec_set=ControllerApplication)
+    app.startup = asynctest.CoroutineMock()
+    app.shutdown = asynctest.CoroutineMock()
+    groups = zigpy.group.Groups(app)
+    groups.add_group(FIXTURE_GRP_ID, FIXTURE_GRP_NAME, suppress_event=True)
+    app.configure_mock(groups=groups)
+    type(app).ieee = mock.PropertyMock()
+    app.ieee.return_value = zigpy.types.EUI64.convert("00:15:8d:00:02:32:4f:32")
+    type(app).nwk = mock.PropertyMock(return_value=zigpy.types.NWK(0x0000))
+    type(app).devices = mock.PropertyMock(return_value={})
+    return app
+
+
+@pytest.fixture
+def zigpy_radio():
+    """Zigpy radio mock."""
+    radio = mock.MagicMock()
+    radio.connect = asynctest.CoroutineMock()
+    return radio
+
+
 @pytest.fixture(name="config_entry")
 async def config_entry_fixture(hass):
     """Fixture representing a config entry."""
-    config_entry = MockConfigEntry(domain=DOMAIN)
-    config_entry.add_to_hass(hass)
-    return config_entry
+    return config_entries.ConfigEntry(
+        version=1,
+        domain=zha_const.DOMAIN,
+        title="Mock Title",
+        data={
+            zha_const.CONF_BAUDRATE: zha_const.DEFAULT_BAUDRATE,
+            zha_const.CONF_RADIO_TYPE: "MockRadio",
+            zha_const.CONF_USB_PATH: "/dev/ttyUSB0",
+        },
+        source="test",
+        connection_class=config_entries.CONN_CLASS_LOCAL_PUSH,
+        system_options={},
+    )
 
 
 @pytest.fixture
-async def setup_zha(hass, config_entry):
-    """Load the ZHA component.
-
-    This will init the ZHA component. It loads the component in HA so that
-    we can test the domains that ZHA supports without actually having a zigbee
-    network running.
-    """
-    # this prevents needing an actual radio and zigbee network available
-    with patch("homeassistant.components.zha.async_setup_entry", async_setup_entry):
-        hass.data[DATA_ZHA] = {}
-
-        # init ZHA
-        await hass.config_entries.async_forward_entry_setup(config_entry, DOMAIN)
-        await hass.async_block_till_done()
-
+def setup_zha(hass, config_entry, zigpy_app_controller, zigpy_radio):
+    """Set up ZHA component."""
+    hass.data[zha_const.DATA_ZHA] = {
+        zha_const.DATA_ZHA_CONFIG: {zha_const.CONF_ENABLE_QUIRKS: False}
```

Comment:
Consider using `async_setup_component` instead, and pass a config that disables the quirk. That will populate the hass.data with the config for us.